include/zck.h.in

```diff
@@ -312,74 +330,74 @@ const char *zck_hash_name_from_type(int hash_type)
-    __attribute__ ((warn_unused_result));
+PUBLIC zckCtx *zck_dl_get_zck(zckDL *dl)
+    ZCK_WARN_UNUSED;
 /* Set zchunk context in download context */
-bool zck_dl_set_zck(zckDL *dl, zckCtx *zck)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_zck(zckDL *dl, zckCtx *zck)
+    ZCK_WARN_UNUSED;
 /* Clear regex used for extracting download ranges from multipart download */
-void zck_dl_clear_regex(zckDL *dl);
+PUBLIC void zck_dl_clear_regex(zckDL *dl);
 /* Download and process the header from url */
-bool zck_dl_get_header(zckCtx *zck, zckDL *dl, char *url)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_get_header(zckCtx *zck, zckDL *dl, char *url)
+    ZCK_WARN_UNUSED;
 /* Get number of bytes downloaded using download context */
-ssize_t zck_dl_get_bytes_downloaded(zckDL *dl)
-    __attribute__ ((warn_unused_result));
+PUBLIC ssize_t zck_dl_get_bytes_downloaded(zckDL *dl)
+    ZCK_WARN_UNUSED;
 /* Get number of bytes uploaded using download context */
-ssize_t zck_dl_get_bytes_uploaded(zckDL *dl)
-    __attribute__ ((warn_unused_result));
+PUBLIC ssize_t zck_dl_get_bytes_uploaded(zckDL *dl)
+    ZCK_WARN_UNUSED;
 /* Set download ranges for zchunk download context */
-bool zck_dl_set_range(zckDL *dl, zckRange *range)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_range(zckDL *dl, zckRange *range)
+    ZCK_WARN_UNUSED;
 /* Get download ranges from zchunk download context */
-zckRange *zck_dl_get_range(zckDL *dl)
-    __attribute__ ((warn_unused_result));
+PUBLIC zckRange *zck_dl_get_range(zckDL *dl)
+    ZCK_WARN_UNUSED;
 
 /* Set header callback function */
-bool zck_dl_set_header_cb(zckDL *dl, zck_wcb func)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_header_cb(zckDL *dl, zck_wcb func)
+    ZCK_WARN_UNUSED;
 /* Set header userdata */
-bool zck_dl_set_header_data(zckDL *dl, void *data)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_header_data(zckDL *dl, void *data)
+    ZCK_WARN_UNUSED;
 /* Set write callback function */
-bool zck_dl_set_write_cb(zckDL *dl, zck_wcb func)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_write_cb(zckDL *dl, zck_wcb func)
+    ZCK_WARN_UNUSED;
 /* Set write userdata */
-bool zck_dl_set_write_data(zckDL *dl, void *data)
-    __attribute__ ((warn_unused_result));
+PUBLIC bool zck_dl_set_write_data(zckDL *dl, void *data)
+    ZCK_WARN_UNUSED;
 
 /* Write callback.  You *must* pass this and your initialized zchunk download
  * context to the downloader when downloading a zchunk file.  If you have your
  * own callback, set dl->write_cb to your callback and dl->wdata to your
  * callback data. */
-size_t zck_write_chunk_cb(void *ptr, size_t l, size_t c, void *dl_v);
-size_t zck_write_zck_header_cb(void *ptr, size_t l, size_t c, void *dl_v);
-size_t zck_header_cb(char *b, size_t l, size_t c, void *dl_v);
+PUBLIC size_t zck_write_chunk_cb(void *ptr, size_t l, size_t c, void *dl_v);
```

Comment:
I should probably remove these, right?

Reply:
My understanding of `PUBLIC` is that it tells the compiler which symbols can exported from the compiled object file, and keeps any of the other symbols private.  I didn't think they were required in `zck.h`, but rather just in the .c files.  Having said all that, I'm not sure how things work with Windows.  Once I'm able to compile in Linux and can verify that the ABI hasn't changed, I'll give you a final answer on whether or not those need to be changed.

Reply:
I just double-checked and these functions are marked public in the `dl.c` source file, so I guess this might just have been an oversight?